### DISCUSSION TITLE
[2024년 03월 12일] 데일리 스크럼

### DISCUSSION THREAD

**Discussion by Subak-Uncle** [cca-ffodregamdi/running-hi-back-v2#41] [dailyscrum]
### 기록

종완

### 📋 회의 내용

* DDD 설계 시에 객체 참조 방법
  - 기존에 사용하던 객체 직접 참조는 도메인 간 Entity 객체 자체를 가져와야 하다 보니 도메인 간 의존성이 발생함.
  - 캡슐화 한 내용을 드러내게 됩니다.
  - 하지만 테이블 의존적인 객체 설계로 인한 '패러다임 불일치' 문제도 있을 것 같다.
-> 우선은, 객체 연관 관계를 이용한 '객체 간 직접 참조' 방법을 이용해 개발 진행!!
  
* GPX 파일 parsing 방법
  - 연산 로직을 어디서 사용할 지.
     -> GPX 파일을 전달 받아 parsing하는 메소드와 함께 service에서 진행!
  - 참고 조사한 자료 공유

- assertJ 버전 관련
  - 기존에 사용하던 테스트 dependency와 호환성 문제 발견
  -> 새 버전인 OO으로 build.gradle에 dependency 추가!
  
* 도메인 분할

### 🐳 오늘까지 한 내용 공유

* 종완 : 북마크 도메인 CRD 개발 완료(merged)
* 수의 : 테스트 코드 작성 및 연구
* 정인 : gpx 파일 연산 메소드 개발 및 조사

### 🦋 알게된 사실 or 참고 자료

* [애그리거트 간에 ID 참조를 해야하는 이유](https://velog.io/@byeongju/%EC%95%A0%EA%B7%B8%EB%A6%AC%EA%B1%B0%ED%8A%B8-%EA%B0%84%EC%97%90-ID-%EC%B0%B8%EC%A1%B0%EB%A5%BC-%ED%95%B4%EC%95%BC%ED%95%98%EB%8A%94-%EC%9D%B4%EC%9C%A0)
* [GPX 파일 형식 참고](https://docs.fileformat.com/ko/gis/gpx/)

### 📌 오늘의 할 일
* 강수의 (CTO) 
  - [x] 피드백 테스트 코드 완성
  - [x] 멤버 시작 티비📺
* 손정인 (MZ 사원, 팀장)
  - [x] Post Create
  - [x] gpx 개념 쫀나 파기
* 김종완 (김인턴)
  - [x] Reply 도메인 CRUD 개발
  - [x]  Test 자동화 CI/CD 설정
  - [ ]  북마크 테스트 코드 작성
* 이효진
  - [x] 자바 쓸 생각에 설레기
  - [x] 프로젝트 환경 설정

**Comment by numerical43** [comment]
왜 팀장님만 오늘 할 일 멀쩡하게 쓰셨어요 가만두지않겠습니다

**Reply by Subak-Uncle** [reply]
헉 호달달,,,, 저는 단지 노션에 적힌 내용을 옮겨 담았습네당ㅠㅠㅠ
